### PR TITLE
Add  MAPIE v1 Regression Integration tests

### PR DESCRIPTION
**Pull request by jawadhussein462**
# Description

This PR adds a set of unit tests to validate the prediction intervals produced by the `v0` and `v1` conformal prediction regressors in MAPIE. These tests ensure interval consistency, coverage, and width alignment across different strategies (`split`, `cv`, `jackknife`, `CQR`) and configurations (confidence levels, methods, conformity scores).

The changes are non-breaking and extend the test suite, adding robustness checks across different model configurations.

Fixes #<issue_number>

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tests were created to run with `pytest`, covering various configurations:
- Strategy types: `split`, `cv`, `jackknife`, `CQR`
- Confidence levels: 0.9, 0.95, 0.99
- Models: `LinearRegression`, `RandomForestRegressor`
- Test set sizes: 0.2, 0.5

- [x] Coverage consistency tests
- [x] Interval width consistency tests
- [x] Basic functionality with a dummy test

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/scikit-learn-contrib/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] Updated [HISTORY.rst](https://github.com/scikit-learn-contrib/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/scikit-learn-contrib/MAPIE/blob/master/AUTHORS.rst)
- [x] Linting: `make lint`
- [x] Typing check: `make type-check`
- [x] Unit tests: `make tests`
- [x] 100% coverage: `make coverage`
- [x] Documentation builds: `make doc`

### CODE REVIEW

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 1]
```diff
@@ -1,9 +1,342 @@
...
+    return np.mean((y_true >= pred_intervals[:, 0]) &
+                   (y_true <= pred_intervals[:, 1]))
+
+
+def initialize_models(
+    strategy_key,
+    v0_params: dict,
+    v1_params: dict,
+    k_folds=5,
+    RANDOM_STATE=42
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py]:
```suggestion
    random_state=42
```

Comment (Valentin-Laurent) [reply mapie_v1/integration_tests/tests/test_regression.py]: Functions params should be lowercase

File: mapie_v1/conformity_scores/utils.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from typing import Union
+from mapie.conformity_scores import (
+    BaseRegressionScore,
+    AbsoluteConformityScore,
+    GammaConformityScore,
+    ResidualNormalisedScore,
+)
+
+
+def select_conformity_score(
```
Comment (Valentin-Laurent) [inline mapie_v1/conformity_scores/utils.py:10]: We definitely need this 👍
Two comments :
- We should probably use a dict instead of if/else statements
- We should put that dict in a __init__ file inside conformity_scores directory

**Review by Valentin-Laurent — COMMENTED** [review]
General comments :
- No comments or docstrings in the code. One exception : when we write code that is hard to make self-explanatory without comments. For example, when using a technique that is not well known by other developers.
- The most important functions should be at the beginning (here: the tests), and the less important at the end (here: initialize_models and calculate coverage)
- We're not testing performance, but only if v1 is exactly or almost exactly the same as v0. So coverage and mean width may be metrics that are too "loose". What about testing that v1 gives the same prediction intervals and the same prediction points as v0 ?
- We miss tests using allow_infinite_bounds or minimize_interval_width

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 3]
```diff
@@ -1,9 +1,334 @@
...
+@pytest.mark.parametrize("confidence_level", [0.9, 0.95, 0.99])
+@pytest.mark.parametrize("agg_function", ["mean", "median"])
+@pytest.mark.parametrize("minimize_interval_width", [True, False])
+@pytest.mark.parametrize("allow_infinite_bounds", [True, False])
+@pytest.mark.parametrize(
+    "estimator", [
+        LinearRegression(),
+        RandomForestRegressor(random_state=RANDOM_STATE, max_depth=2)])
+@pytest.mark.parametrize("test_size", [0.2, 0.5])
+def test_exact_interval_equality(
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py:47]: Let's have 1 test for each v1 class maybe, so that one would be `test_exact_interval_equality_for_split`

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 4]
```diff
@@ -1,9 +1,334 @@
...
+
+@pytest.mark.parametrize("strategy_key", ["split"])
+@pytest.mark.parametrize("method", ["base", "plus", "minmax"])
+@pytest.mark.parametrize("conformity_score", ["absolute"])
+@pytest.mark.parametrize("confidence_level", [0.9, 0.95, 0.99])
+@pytest.mark.parametrize("agg_function", ["mean", "median"])
+@pytest.mark.parametrize("estimator", [LinearRegression(),
+                                       RandomForestRegressor(max_depth=2)])
+@pytest.mark.parametrize("test_size", [0.2, 0.5])
+def test_consistent_coverage(
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py:121]: `test_consistent_coverage_for_split`

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 5]
```diff
@@ -1,9 +1,334 @@
...
+    tolerance = 0.06
+    err_msg = (
+        f"Normalized interval width difference too high: "
+        f"v0 avg width {v0_avg_width}, v1 avg width {v1_avg_width}, "
+        f"normalized difference {normalized_difference}"
+    )
+    assert normalized_difference < tolerance, err_msg
 
 
 def test_dummy():
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py:330]: We can delete that one

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 6]
```diff
@@ -1,9 +1,334 @@
...
+@pytest.mark.parametrize("strategy_key", ["split"])
+@pytest.mark.parametrize("method", ["base", "plus", "minmax"])
+@pytest.mark.parametrize("conformity_score", ["absolute"])
+@pytest.mark.parametrize("confidence_level", [0.9, 0.95, 0.99])
+@pytest.mark.parametrize("agg_function", ["mean", "median"])
+@pytest.mark.parametrize("estimator", [
+        LinearRegression(),
+        RandomForestRegressor(random_state=RANDOM_STATE, max_depth=2)])
+@pytest.mark.parametrize("test_size", [0.2, 0.5])
+def test_consistent_interval_width(
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py:256]: `test_consistent_interval_width_for_split`

**Review by Valentin-Laurent — COMMENTED** [review]
General comments:
 - No dosctrings 
 - Why do we test only `absolute` score?
 - For now, `train_test_split_shuffle` and `filter_params` are very specific to integration_tests, so I would put the `utils.py` in the integration_tests folder.
 - Duplicated code (we can address that later on)
